src/electron/main-process/open-with-temp-file.ts

```diff
@@ -1,11 +1,30 @@
 import {app, shell} from 'electron';
-import {writeFile} from 'fs-extra';
+import { writeFile, remove } from 'fs-extra';
 import path from 'path';
 import uuid from 'tiny-uuid';
 
-export async function openWithTempFile(data: string, suffix: string) {
-	const tempPath = path.join(app.getPath('temp'), uuid() + suffix);
+let oldtmp:string|null = null;
 
+function deleteTempFile(path: string|null, required_subroot: string) {
+	if (path != null && path.startsWith(required_subroot)) {
```

Comment:
This check trades speed for safety, preventing the accidental deletion of non-temporary files if the location of the temp hierarchy is forcefully changed by the sysadmin during runtime.
This should allow you to reuse this code elsewhere with confidence.